File: Data Structures & Algorithms/binary-search/submission-0.cpp
class Solution {
public:
    int binsearch(vector<int>&nums, int low, int high, int tgt)
    {
        while(low<=high)
        {
            int mid = (low+high)/2;
            if(nums[mid]==tgt) return mid;
            else if(nums[mid]>tgt)
            {
                high=mid-1;
            }else{
                low=mid+1;
            }
        }
        return -1;
    }
    int search(vector<int>& nums, int target) {
        int n = nums.size();
        int low = 0, high = n-1;
        return binsearch(nums,low,high,target);
    }
};

File: Data Structures & Algorithms/divide-array-into-equal-pairs/submission-0.java
class Solution {
    public boolean divideArray(int[] nums) {
        Arrays.sort(nums);
        int n = nums.length / 2;
        int i=0;
        
        while(i<nums.length)
        {
            if(nums[i]!=nums[i+1])return false;
            i+=2;
        }
        return true;
    }
}